File: 图论/200. 岛屿数量.cpp
#include <vector>
using namespace std;
class Solution
{
    public:
        int numIsLands(vector<vector<char>>&grid)
        {
            int n=grid.size();
            int m=grid[0].size();
            vector<vector<bool>>visited=vector<vector<bool>>(n,vector<bool>(m,false));
            int result=0;
            for(int i=0;i<n;i++){
                for(int j=0;j<m;j++){
                    if (!visited[i][j]&& grid[i][j]=='1')
                    {
                        visited[i][j]=true;
                        dfs(grid,visited,i,j);
                    }
                }
            }
        }
    private:
        int dirs[4][2]={0,1,0,-1,1,0,-1,0};
        void dfs(vector<vector<char>>& grid,vector<vector<bool>>& visited,int x, int y){
            for(int i=0;i<4;i++)
            {
                int nextx=x+dirs[i][0];
                int nexty=y+dirs[i][1];
                if(nextx<0 ||nextx>=grid.size()||nexty<0||nexty>=grid[0].size()){continue;}
                
            }
    }
};

File: 算法篇/深度优先搜索 DFS/200. 岛屿数量.py
class Solution(object):
    def numIslands(self, grid):
        """
        :type grid: List[List[str]]
        :rtype: int
        """        

        res=0
        for row in range(len(grid)):
            for col in range(len(grid[0])):
                if grid[row][col]=="1":
                    self.dfs(grid,row,col)
                    res += 1
        return res

    def dfs(self,grid,i,j):
        dirs=[[-1,0],[1,0],[0,-1],[0,1]]
        grid[i][j]="0"
        for dir in dirs:
            nrow,ncol=i+dir[0],j+dir[1]
            if nrow>=0 and nrow<len(grid) and ncol>=0 and ncol<len(grid[0]):
                if grid[nrow][ncol]=="1":
                    self.dfs(grid,nrow,ncol)
        
solution=Solution()
grid=[["1","1"],["1","1"]]
res=solution.numIslands(grid) 
print(res)           
